internal_task_pollers.go

```diff
@@ -587,3 +588,22 @@ func convertActivityResultToRespondRequest(identity string, taskToken, result []
 		Details:   details,
 		Identity:  common.StringPtr(identity)}
 }
+
+func convertActivityResultToRespondRequestByID(identity, domainID, workflowID, activityID string,
```

Comment:
this should call a correspondent service API. The token should be opaque to the clients.

Reply:
Updated